### PR TITLE
Allow to set CSS class via plone.autoform directives

### PR DESCRIPTION
**Pull request by thet**
- Set the ``firstDay`` option on the ``DatetimeWidget`` likewise as ``DateWidget``.
- Don't overwrite widget default css classes when rendering pattern widgets.
  This allows setting a css class via the ``klass`` keyword in plone.autoform widget directives.

### CODE REVIEW

File: plone/app/z3cform/tests/test_widgets.py [hunk 1]
```diff
@@ -201,20 +217,22 @@ def setUp(self):
...
         self.widget.pattern_options = {
-            'date': {'firstDay': 0},
             'time': {'interval': 15}
         }
 
+    @mock.patch('plone.app.z3cform.widget.first_weekday', new=lambda: 3)
     def test_widget(self):
         current_year = datetime.today().year
+        import ipdb
+        ipdb.set_trace()
```
Comment (gforcada) [inline plone/app/z3cform/tests/test_widgets.py]: Oops

File: plone/app/z3cform/widget.py [hunk 2]
```diff
@@ -90,7 +90,14 @@ def render(self):
         """
         if self.mode != 'input':
             return super(BaseWidget, self).render()
-        return self._base(**self._base_args()).render()
+
+        pattern_widget = self._base(**self._base_args())
+        if getattr(self, 'klass', False):
+            pattern_widget.klass = u'{0}{1}'.format(
+                pattern_widget.klass,
+                u' ' + self.klass
```
Comment (gforcada) [inline plone/app/z3cform/widget.py]: why don't you set it like: ```u'{0} {1}'.format(patter_widget.klass, self.klass)``` ?

Comment (thet) [inline plone/app/z3cform/widget.py]: Ah true - I did not want an extra space if ``self.klass`` is None, but that's never the case with the ``if getattr`` above. tnx!

File: plone/app/z3cform/tests/test_widgets.py (hunk 1 above)
Comment (thet) [inline plone/app/z3cform/tests/test_widgets.py]: Ooops! :)
That's the dev version which wasn't supposed to be checked in. It even has some changes to other tests, which are not part of this PR. Tnx for the pointer!
